### PR TITLE
drivers: serial: add dts peripherals to stm32g0

### PR DESCRIPTION
**Pull request by kiram9**
Added device tree nodes and associated headers for
defined uarts on the stm32g0 and stm32g07x 8x parts.

Tested with uart on stm32g071rb disco board with usart3 going to stlink.
Using shell.

needs https://github.com/zephyrproject-rtos/zephyr/pull/23657 pinmux

Signed-off-by: Kieran Levin <ktl@frame.work>

### CODE REVIEW

**Review by FRASTM — APPROVED** [review]

File: dts/arm/st/g0/stm32g071.dtsi [hunk 1]
```diff
@@ -6,3 +6,27 @@
...
+/ {
+	soc {
+		usart3: serial@40004800 {
+				compatible = "st,stm32-usart", "st,stm32-uart";
+				reg = <0x40004800 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00040000>;
+				interrupts = <29 0>;
+				interrupt-names = "combined";
+				status = "disabled";
+				label = "UART_3";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g071.dtsi]: Please fix indentation

**Review by erwango — CHANGES_REQUESTED** [review]
Nit pick.

File: dts/arm/st/g0/stm32g071.dtsi (hunk 1 above)
Comment (kiram9) [reply dts/arm/st/g0/stm32g071.dtsi]: fixed

**Review by erwango — APPROVED** [review]
Approved, but please clean up the commit message.
The following has interest in scope of PR review, but provide no interest in the tree once merged.
```needs #23657 pinmux
for alternate pin function selection.

fixed indentation```

**Pull request merged**